### ISSUE TITLE
Monster Box on top of UI

### ISSUE THREAD

**Issue by LordPyrex** [NPBruce/valkyrie#506]
Is there a way for UI elements to build on the layer above the monster box? I've been handling this for end game scenario to simply remove all those assets, but then noticed that the one mid game scene that I wrote would eliminate a live spawn. Also is there a command that would remove all monsters without having to set them all as remove components?

**Comment by cbusbird** [comment]
I think you can remove all monsters by selecting a var from the "Special" selection that comes up when you choose to remove a component.  I just noticed this myself and haven't tried it yet but I assume that's what it is for.

**Comment by cbusbird** [comment]
There is #monsters and #boardcomponents, I would assume that removing one of these would remove all active spawns or all components that are on the board currently.  I will be trying out the #boardcomponents shortly here and will report back.

**Comment by LordPyrex** [comment]
#monsters is a variable for number of monsters and same with board
components. They represent a usable number to trigger events.

**Comment by cbusbird** [comment]
I just tested and if you remove #boardcomponents it does remove all the tokens and tiles that have already been replaced on the board.

**Comment by cbusbird** [comment]
It's perfect for a situation where you clear the board and start a new map, something a few of the FFG scenarios do.  I'm positive it will work for #monsters as well.

**Comment by LordPyrex** [comment]
Intrresting..... that's not how it was listed in the variables page. I'll
have to check that and do some wiki updates.

**Comment by cbusbird** [comment]
Just tested for #monsters and it did remove all active spawns.  I'm pretty sure it's a newer feature.  I was working last night and suddenly noticed the "Special" section that shows up when you go the the "Remove Components" selection screen.

I haven't even touch the UI stuff yet.  I figure I will do those parts as finishing touches on my scenario.  But I'm sure Bruce will have an answer.

**Comment by cbusbird** [comment]
Is exotic material your scenario?

**Comment by LordPyrex** [comment]
No, I will be rolling mine out some time in the next week when I finish
activations, horror checks, and evades for my special spawn. Exotic is
Bruce's.

**Comment by cbusbird** [comment]
Cool, can't wait to try it.  Mine will probably be a little longer than a week until its ready.  There's a lot more work involved than I initially thought for putting one of these together!  But I'm really enjoying working on during my free time.

**Comment by NPBruce** [comment]
In this case it is not a var (you can't remove var).  From https://github.com/NPBruce/valkyrie/wiki/QuestIniEvent#remove :

If #boardcomponents is used all tokens doors, UIs and tiles will be removed. If #monsters is used all monsters will be removed (without causing the defeated triggers). If #shop is used the item shop will be removed.

Note that another option is to use 2 quests.  You can have one quest start another as an event, which will wipe all of these things and may be easier to manage for very large scenarios to break them up.  If the second quest is marked as 'hidden' it won't be startable directly.  This is how the demos work.

**Comment by cbusbird** [comment]
I have recently started playing with some UI elements on my scenario and have run into what @LordPyrex was seeing.  If you have an active monster when you call the UI components the monster icon remains on top of the UI.  Using the remove #monsters does resolve this, but in my case I want to implement a UI element during the middle of the scenario.  This event can happen when monsters are active and I don't want to remove them permanently.  I would like the UI elements to lay on top of the monster icons so that they remain active but aren't screwing up the vibe of the UI.

![monsterontopofui](https://user-images.githubusercontent.com/28448698/27012019-2e3289de-4e96-11e7-8529-0742e373228c.PNG)

**Comment by NPBruce** [comment]
I will think about what I can do.
